src/engraving/rendering/score/systemlayout.cpp

```diff
@@ -1432,9 +1432,9 @@ void SystemLayout::layoutSystemElements(System* system, LayoutContext& ctx)
             }
 
             staff_idx_t si = e->staffIdx();
-            Segment* s = toSegment(parent);
```

Comment:
Here, it looks like the real solution would be to eliminate `parent` and the local `s`, and use the `s` from the `for` loop instead: `e` is one of the `annotations` of the `s` from the `for` loop, so `e->parentItem` is supposed to be `s` anyway.

Reply:
You mean loke this?
```cpp
            staff_idx_t si = e->staffIdx();
            Measure* m = s->measure();
            system->staff(si)->skyline().add(e->shape().translate(e->pos() + s->pos() + m->pos() + e->staffOffset()));
```

Reply:
Yes, but I think you can additionally remove `EngravingItem* parent` and the `IF_ASSERT_FAILED` that follows it, because that's also not necessary anymore.

Reply:
II see. Done